qa/integration/services/logstash_service.rb

```diff
@@ -202,24 +206,54 @@ def application_settings_file
   end
 
   def plugin_cli
-    PluginCli.new(@logstash_home)
+    PluginCli.new(self)
   end
 
   def lock_file
     File.join(@logstash_home, "Gemfile.lock")
   end
 
+  def run_cmd(cmd_parameters, change_dir = true, environment = {})
+    out = Tempfile.new("content")
+    out.sync = true
+
+    parts = cmd_parameters.split(" ")
+    cmd = parts.shift
+
+    process = ChildProcess.build(cmd, *parts)
+    environment.each do |k, v|
+      process.environment[k] = v
+    end
+    process.io.stdout = process.io.stderr = out
+
+    Bundler.with_clean_env do
+      if change_dir
+        Dir.chdir(@logstash_home) do
+          process.start
+        end
+      else
+        process.start
+      end
+    end
+
+    process.poll_for_exit(TIMEOUT_MAXIMUM)
+    out.rewind
+    ProcessStatus.new(process.exit_code, out.read)
+  end
+
+  def run(command)
+    run_cmd("#{@logstash_bin} #{command}")
```

Comment:
should we just pass an array here since the one of the first things we do in `run_cmd` is again split by space?
```suggestion
    run_cmd([@logstash_bin, command])
```

Reply:
thanks, refactored (was a :scissors: so I did not change much around the original piece).
kept `run_raw` with a `cmd.split(' ')` as that one is used by other integration specs.